cluster/ubuntu/util.sh

```diff
@@ -430,6 +440,24 @@ function provision-node() {
     ubuntu/binaries/minion \
     "${1}:~/kube"
 
+  # copy the flannel binaries and scripts, if flannel is being used
+  if [ -z "$CNI_PLUGIN_CONF" ] || [ -z "$CNI_PLUGIN_EXES" ]; then
+    scp -r $SSH_OPTS ubuntu/minion-flannel/* "${1}:~/kube"
+  fi
+
+  # copy the CNI config and binaries, if being used, to the node
+  if [ -n "$CNI_PLUGIN_CONF" ] && [ -n "$CNI_PLUGIN_EXES" ]; then
+    ssh $SSH_OPTS "${1}" "rm -rf tmp-cni; mkdir -p tmp-cni/exes tmp-cni/conf"
+    scp    $SSH_OPTS "$CNI_PLUGIN_CONF" "${1}:tmp-cni/conf/"
+    scp -p $SSH_OPTS  $CNI_PLUGIN_EXES  "${1}:tmp-cni/exes/"
+    ssh $SSH_OPTS -t "${1}" '
+      sudo -p "[sudo] password to prep node %h: " -- /bin/bash -ce "
+        mkdir -p /opt/cni/bin /etc/cni/net.d
+        cp ~$(id -un)/tmp-cni/conf/* /etc/cni/net.d/
+        cp --preserve=mode ~$(id -un)/tmp-cni/exes/* /opt/cni/bin/
+        "'
+  fi
```

Comment:
Isn't all of this stuff that belongs in salt or something...   I might not like salt, but doing sysadmin like this in bash is just terrible terrible terrible....